sklearn/utils/estimator_checks.py

```diff
@@ -3543,7 +3543,11 @@ def _enforce_estimator_tags_y(estimator, y):
         # y could be of integer dtype.
         y += 1 + abs(y.min())
     # Estimators with a `binary_only` tag only accept up to two unique y values
-    if _safe_tags(estimator, key="binary_only") and y.size > 0:
+    # TargetEncoder accepts binary and continuous values. We therefore force
+    # y to be binary also.
```

Comment:
can that be through an estimator tag rather than hard coding here? making encoders pass common tests shouldn't be fixing common tests to accept encoders lol

Reply:
Yep, but this is a really weird thing: a transformer that uses `y` and can be classification and regression but only binary classification. This is just a beast.

However, I special case with another comment because we should be dropping this issue when merging support for multiclass: https://github.com/scikit-learn/scikit-learn/pull/26674

Reply:
Do you think I should do this in https://github.com/scikit-learn/scikit-learn/pull/26674?

Reply:
#26674 should modify the code such that the test is passing. I assume I cannot do the trick of @thomasjpfan using `binary_only` and you will remove it in your PR.